src/System.DirectoryServices/tests/System/DirectoryServices/ActiveDirectory/DomainControllerTests.cs

```diff
@@ -39,10 +39,10 @@ public void GetDomainController_InvalidName_ThrowsActiveDirectoryObjectNotFoundE
 
         [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
         [SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "Access to path is denied when in App container")]
-        public void GetDomainController_InvalidIPV6_ThrowsInvalidCastException()
+        public void GetDomainController_InvalidIPV6_ThrowsActiveDirectoryObjectNotFoundException()
         {
             var context = new DirectoryContext(DirectoryContextType.DirectoryServer, "[::1]:port");
-            Assert.Throws<InvalidCastException>(() => DomainController.GetDomainController(context));
+            Assert.Throws<ActiveDirectoryObjectNotFoundException>(() => DomainController.GetDomainController(context));
```

Comment:
On a non-joined machine this is fast, they do take time when the machine is domain joined. I have to measure the difference between domain joined connected and disconnected from the domain. I would like to make the tests pass on my main dev box that is domain joined when doing that we may want to skip some tests that can take a long time (~30 seconds).